plugins/services/src/js/components/ServiceStatusWarning.js

```diff
@@ -7,29 +7,48 @@ import Icon from '../../../../../src/js/components/Icon';
 import Pod from '../structs/Pod';
 import Service from '../structs/Service';
 import ServiceTree from '../structs/ServiceTree';
+import StringUtil from '../../../../../src/js/utils/StringUtil';
 
 class ServiceStatusWarning extends Component {
+  getTooltip(content) {
+    return (
+      <Tooltip
+        content={content}
+        width={250}
+        wrapText={true}
+        wrapperClassName="tooltip-wrapper status-waiting-indicator">
+        <Icon color="red" id="yield" size="mini" />
+      </Tooltip>
+    );
+  }
+
   getTooltipContent(timeWaiting) {
     return `DC/OS has been waiting for resources and is unable to complete this deployment for ${DateUtil.getDuration(timeWaiting, null)}.`;
   }
+
   render() {
     const {item} = this.props;
-    const queue = item.getQueue();
-
-    if (queue != null
-      && DeclinedOffersUtil.shouldDisplayDeclinedOffersWarning(queue)) {
-      const waitingSince = DateUtil.strToMs(queue.since);
-      const timeWaiting = Date.now() - waitingSince;
-
-      return (
-        <Tooltip
-          content={this.getTooltipContent(timeWaiting)}
-          width={250}
-          wrapText={true}
-          wrapperClassName="tooltip-wrapper status-waiting-indicator">
-          <Icon color="red" id="yield" size="mini" />
-        </Tooltip>
-      );
+
+    if (item instanceof ServiceTree) {
+      const appsWithDeclinedOffers = DeclinedOffersUtil
+        .getAppsWithDeclinedOffersFromServiceTree(item);
+      const appCount = appsWithDeclinedOffers.length;
+
+      if (appCount > 0) {
+        const appsNoun = StringUtil.pluralize('service', appCount);
+
+        return this.getTooltip(`DC/OS is waiting for resources and is unable to complete the deployment of ${appCount} ${appsNoun} in this group.`);
+      }
+    } else {
+      const queue = item.getQueue();
+
+      if (queue != null
+        && DeclinedOffersUtil.shouldDisplayDeclinedOffersWarning(queue)) {
+        const waitingSince = DateUtil.strToMs(queue.since);
```

Comment:
Don't we need same calculation as here? https://github.com/dcos/dcos-ui/pull/1721/files#diff-05d538b8dad1163c5f63426574d0d46aR223